lib/components/narrative/line-itin/connected-itinerary-body.js

```diff
@@ -61,6 +62,30 @@ class ConnectedItineraryBody extends Component {
     } = this.props
     const { LegIcon } = this.context
 
+    let [pickupAdded, dropOffAdded] = [false, false]
+
+    // Support OTP1 flex messages in Trip Details
+    itinerary.legs = itinerary.legs.map((oldLeg) => {
```

Comment:
I would rather not touch the `itinerary` prop here. Maybe instead of cloning each leg as below, clone the entire `prop.itinerary` and modify the cloned itinerary at leisure (no need for a `.map()` or `clone`ing the legs in that case, a simple `.forEach()` will do). Then pass the resulting itinerary to `<StyledItineraryBody>` below.

Reply:
whole itinerary cloned in https://github.com/opentripplanner/otp-react-redux/pull/571/commits/dd2eb27dda8441bac039f6bc0095269df86c09cf